airbyte-integrations/connectors/source-s3/source_s3/s3file.py

```diff
@@ -59,8 +59,15 @@ def open(self, binary: bool) -> Iterator[Union[TextIO, BinaryIO]]:
             config = ClientConfig(signature_version=UNSIGNED)
             params = {"client": make_s3_client(self._provider, config=config)}
         self.logger.debug(f"try to open {self.file_info}")
-        result = smart_open.open(f"s3://{bucket}/{self.url}", transport_params=params, mode=mode)
-
+        # There are rare cases when some keys become unreachable during sync
+        # and we don't know about it, because catalog has been initially formed only once at the beginning
+        # This is happen for example if a file was deleted/moved (or anything else) while we proceed with another file
+        try:
+            result = smart_open.open(f"s3://{bucket}/{self.url}", transport_params=params, mode=mode)
+        except OSError as e:
+            self.logger.warn(f"We don't have access to {self.url}. "
+                             f"Check whether key {self.url} exists in `{bucket}` bucket and/or has proper ACL permissions")
+            raise e
```

Comment:
Does this produce an AirbyteTraceError message?  Just logging to stdout won't help users learn what to do

Reply:
cc @brianjlai maybe you recall the answer - does throwing an exception here end up making a useable AirbyteTraceErrorMessage?

Reply:
It does. But in this case I catch it in two possible places where it can be raised and handle it

Reply:
Ok, this matches the philosophy that @erohmensing is working on with streams being unavailable - we sync what we can and log an error for the streams we can't reach.  👍